frontend/lib/search/result_list.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:math';
+
+import 'package:ehrenamtskarte/graphql/graphql_api.graphql.dart';
+import 'package:ehrenamtskarte/map/detail/detail_view.dart';
+import 'package:ehrenamtskarte/widgets/error_message.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:graphql_flutter/graphql_flutter.dart';
+
+class ResultList extends StatelessWidget {
+  final String _searchText;
+  final List<int> _searchCategories;
+
+  ResultList(this._searchText, this._searchCategories);
+
+  @override
+  Widget build(BuildContext context) {
+    final byIdQuery = AcceptingStoresSearchQuery(
+        variables: AcceptingStoresSearchArguments(
+            params: SearchParamsInput(
+                categoryIds: _searchCategories, searchText: _searchText)));
+    return Query(
+      options: QueryOptions(
+          documentNode: byIdQuery.document,
+          variables: byIdQuery.getVariablesMap()),
+      builder: (QueryResult result,
+          {VoidCallback refetch, FetchMore fetchMore}) {
+        if (result.hasException) {
+          return SliverToBoxAdapter(
+              child: ErrorMessage(result.exception.toString()));
+        }
+
+        if (result.loading) {
+          return SliverToBoxAdapter(child: LinearProgressIndicator());
+        }
+        final matchingStores = AcceptingStoresSearchQuery()
+            .parse(result.data)
+            .searchAcceptingStores;
+        if (matchingStores.isEmpty) {
+          return SliverToBoxAdapter(child: Text("Keine Ergebnisse gefunden"));
+        }
+        return SliverList(
+            delegate: SliverChildBuilderDelegate(
+          (BuildContext context, int index) {
+            final int itemIndex = index ~/ 2;
+            if (index.isEven) {
+              return ListTile(
+                  title: Text(matchingStores[itemIndex].name),
+                  subtitle: Text(
+                    matchingStores[itemIndex].description,
+                    maxLines: 2,
+                    overflow: TextOverflow.ellipsis,
+                  ),
+                  isThreeLine: true,
+                  trailing: Container(
+                      child: Icon(Icons.keyboard_arrow_right, size: 30.0),
+                      height: double.infinity),
+                  onTap: () {
+                    _openDetailView(context, matchingStores[itemIndex].id);
+                  });
+            }
+            return Divider(height: 0, color: Colors.grey);
+          },
+          semanticIndexCallback: (Widget widget, int localIndex) {
+            if (localIndex.isEven) {
+              return localIndex ~/ 2;
+            }
+            return null;
+          },
+          childCount: max(0, matchingStores.length * 2 - 1),
+        ));
```

Comment:
Maybe we could extract this into a separate build function or maybe even a separate widget to reduce the complexity of this build function.

Reply:
I extracted the list